framework/src/utils/NestedSolve.C

```diff
@@ -11,66 +11,94 @@
 
 #include "libmesh/utility.h"
 
+template <bool is_ad>
 InputParameters
-NestedSolve::validParams()
+NestedSolveTempl<is_ad>::validParams()
 {
   InputParameters params = emptyInputParameters();
 
   // Newton iteration control parameters
   params.addParam<Real>("relative_tolerance",
-                        NestedSolve::relativeToleranceDefault(),
+                        relativeToleranceDefault(),
                         "Relative convergence tolerance for Newton iteration");
   params.addParam<Real>("absolute_tolerance",
-                        NestedSolve::absoluteToleranceDefault(),
+                        absoluteToleranceDefault(),
                         "Absolute convergence tolerance for Newton iteration");
+  params.addParam<Real>(
+      "x_tolerance", xToleranceDefault(), "Threshold for minimum step size of linear iterations");
```

Comment:
why x? x for linear?

Reply:
that seems to be a fairly common term. it is `x` as in `f(x)=0`, this is the tolerance for changes in the solution. If the solution change is less than this value we assume the iterations to be converged.

Reply:
x = f(x) is the nonlinear problem ?